### PR TITLE
MeshCutter: Can set `planeNormal/Distance`, and add `isIndexed`.

### PR DESCRIPTION
**Pull request by gonnavis**
Sibling PRs: https://github.com/webaverse/app-wasm/pull/54 , https://github.com/webaverse/mesh-cut-example/pull/3

MeshCutter arg: Use `planeNormal/Distance` instead of `planeQuaternion`, and can set it by needs now.

And add arg: `isIndexed`, to support both indexed and non-indexed geometries, in order to do recursively cutting.

image

image

image

### CODE REVIEW

File: physx.js [hunk 1]
```diff
@@ -891,17 +892,14 @@ const physxWorker = (() => {
     const uvsTypedArray = allocator.alloc(Float32Array, numUvs)
     uvsTypedArray.set(uvs)
 
-    const facesTypedArray = allocator.alloc(Uint32Array, numFaces)
-    facesTypedArray.set(faces)
-
-    const positionTypedArray = allocator.alloc(Float32Array, 3)
-    positionTypedArray.set(position)
+    let facesTypedArray;
```
Comment (avaer) [inline physx.js:895]: Btw this could be done with just passing `faces === null`.

File: physx.js [hunk 2]
```diff
@@ -891,17 +892,14 @@ const physxWorker = (() => {
     const uvsTypedArray = allocator.alloc(Float32Array, numUvs)
     uvsTypedArray.set(uvs)
 
-    const facesTypedArray = allocator.alloc(Uint32Array, numFaces)
-    facesTypedArray.set(faces)
-
-    const positionTypedArray = allocator.alloc(Float32Array, 3)
-    positionTypedArray.set(position)
+    let facesTypedArray;
+    if (isIndexed) {
```
Comment (avaer) [inline physx.js:896]: Btw this could be done with just passing faces === null.

**Review by avaer — APPROVED** [review]
Looks good, just one cleanup I would have recommended.

File: physx.js (hunk 2 above)
Comment (gonnavis) [reply physx.js:896]: Modified https://github.com/webaverse/app/pull/2954
At js side, I check if `faces` is falsy, is it ok? Or need to input exact 'null' ?

**Pull request merged**